lib/web/apiserver.go

```diff
@@ -365,6 +365,10 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
 	h.GET("/webapi/sites/:site/nodes/:server/:login/scp", h.WithClusterAuth(h.transferFile))
 	h.POST("/webapi/sites/:site/nodes/:server/:login/scp", h.WithClusterAuth(h.transferFile))
 
+	// Sign required files to setup mTLS in other services (eg DBs)
+	// POST /webapi/sites/:site/sign
+	h.POST("/webapi/sites/:site/sign", h.WithProvisionTokenAuth(h.signCertKeyPair, types.RoleDatabase))
```

Comment:
This makes this API call specific to signing database certificates which would make it challenging to reuse for other formats later (if need arises). May not be a big deal but I was thinking, can we call WithProvisionTokenAuth after we already know the format the user is requesting? Then we can see, if the user requests one of database formats, we'd check for "db" role in the token, in future if the user requests SSH format, we'd check for "node" role, etc.

This would mean we probably can't wrap the handler function here but rather should call WithProvisionTokenAuth explicitly in the handler (`signCertKeyPair`) after parsing the request parameters.

Reply:
I agree this cannot be easily extendable to other formats
So, let's focus on the `db` format

For now, I'll change the url path to `/webapi/sites/:site/sign/db` and be explicit about this being exclusively for database certificates.

Later on, we can change the path to be
`/webapi/sites/:site/sign/:format` and route the request to a specific handler based on the `format` path variable

Let me know what do you think

Reply:
Yeah this works as well.